src/runtime/failure.rs

```diff
@@ -0,0 +1,52 @@
+use std::fs::OpenOptions;
+use std::io::{ErrorKind, Write};
+use std::path::{Path, PathBuf};
+
+use crate::scheduler::serialization::serialize_schedule;
+use crate::scheduler::Schedule;
+use crate::{Config, FailurePersistence};
+
+/// Produce a message describing how to replay a failing schedule.
+pub fn persist_failure(schedule: &Schedule, message: String, config: &Config) -> String {
+    let serialized_schedule = serialize_schedule(schedule);
+    // Try to persist to a file, but fall through to stdout if that fails for some reason
+    if let FailurePersistence::File { directory } = &config.failure_persistence {
+        let rslt = persist_failure_to_file(&serialized_schedule, directory.as_ref());
+        if let Ok(path) = rslt {
+            return format!("{}\nfailing schedule persisted to file: {}\npass that path to `shuttle::replay_from_file` to replay the failure", message, path.display());
+        }
```

Comment:
How about printing an error message to warn that writing to the file failed, and we're falling back to console?